index/scorch/builder.go

```diff
@@ -106,7 +106,7 @@ func (o *Builder) parseConfig(config map[string]interface{}) (err error) {
 	if forcedSegmentType != "" && forcedSegmentVersion != 0 {
 		segPlugin, err := chooseSegmentPlugin(forcedSegmentType,
 			uint32(forcedSegmentVersion))
-		if err != nil {
+		if err == nil {
```

Comment:
good catch!
Errs is returned on line no:104 and above, but here at line no:109 error is ignored?